src/libawkward/array/ListOffsetArray.cpp

```diff
@@ -1872,19 +1860,18 @@ namespace awkward {
 
       ContentPtr outcontent = trimmed.get()->sort_next(
         negaxis, util::make_starts(offsets_), nextparents,
-        offsets_.length() - 1, ascending, stable, false);
+        offsets_.length() - 1, ascending, stable);
+
+      Index64 outoffsets(offsets_.length());
+      // FIXME: move to kernel
+      for (int64_t i = 0; i < offsets_.length(); i++) {
+        outoffsets.data()[i] = offsets_.data()[i] - offsets_.data()[0];
+      }
```

Comment:
@agoose77 - same here: a temporary fix - I left it for you to provide a proper one :-)